ide/project.dependency/src/org/netbeans/modules/project/dependency/DependencyResult.java

```diff
@@ -45,8 +46,18 @@ public interface DependencyResult extends Lookup.Provider {
      */
     public Dependency getRoot();
     
+    /**
+     * Checks if the data is still valid
+     * @return true, if the data is valid
+     */
     public boolean isValid();
     
+    /**
+     * Returns artifacts that may be unavailable or erroneous.
+     * @return problem artifacts
+     */
+    public Collection<ArtifactSpec> getProblemArtifacts();
+    
```

Comment:
So this artefacts are not part of the dependency tree?

Reply:
They can be ... but they are 'problematic' :) e.g. they do not have the `getFile()` valid / filled in. Or some other issue like version conflict. Maybe this could rather identify the `Dependency` structure -- which means an exact position in the tree  ?